pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -240,65 +253,64 @@ func (h *calicoIPPoolHandler) updateROKSCalicoCfg(ctx context.Context) error {
 		return nil
 	}
 
-	// platform is ROKS, make sure that IPIPMode of default IPPool is Always
-
-	err = util.Update(ctx, h.iPPoolResourceInterface(), &calicoapi.IPPool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: DefaultV4IPPoolName,
-		},
-	}, func(existing *calicoapi.IPPool) (*calicoapi.IPPool, error) {
-		if existing.Spec.IPIPMode == calicoapi.IPIPModeAlways {
-			logger.Infof("IPIPMode of %s IPPool already set to Always", DefaultV4IPPoolName)
-			return existing, nil // no need to update
-		}
+	// platform is ROKS, make sure that encapsulation of default Installation is set to IPIP
+	// In this way the Tigera Operator will set IPIPMode of default IPPool to Always
+	return util.Update(ctx, resource.ForDynamic(h.dynClient.Resource(InstallationsGVR)),
+		resource.MustToUnstructured(&tigerav1.Installation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: DefaultInstallationName,
+			},
+		}), func(existing *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+			installation := &tigerav1.Installation{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(existing.Object, installation)
+			utilruntime.Must(err)
 
-		if existing.Labels == nil {
-			existing.Labels = make(map[string]string)
-		}
+			if installation.Spec.CalicoNetwork == nil {
+				installation.Spec.CalicoNetwork = &tigerav1.CalicoNetworkSpec{}
+			}
 
-		if existing.Annotations == nil {
-			existing.Annotations = make(map[string]string)
-		}
+			ipPools := installation.Spec.CalicoNetwork.IPPools
+			if len(ipPools) > 0 && ipPools[0].Encapsulation == tigerav1.EncapsulationIPIP {
+				logger.Infof("Encapsulation of " + DefaultInstallationName + " Installations already set to IPIP")
+				return existing, nil
+			}
 
-		// mark 'submariner-managed' label for persistency
-		existing.Labels[submarinerManagedLabel] = "true"
-		existing.Annotations[submarinerPrevIPIPMode] = string(existing.Spec.IPIPMode)
-		existing.Spec.IPIPMode = calicoapi.IPIPModeAlways
+			if installation.Annotations == nil {
+				installation.Annotations = map[string]string{}
+			}
 
-		logger.Infof("Setting IPIPMode of %s IPPool to Always", DefaultV4IPPoolName)
+			installation.Annotations[submarinerPrevEncapsulation] = ipPools[0].Encapsulation.String()
+			installation.Annotations[submarinerManagedLabel] = "true"
 
-		return existing, nil
-	})
+			ipPools[0].Encapsulation = tigerav1.EncapsulationIPIP
```

Comment:
This will panic if `ipPools` is empty.

Reply:
Thx. Added a precheck and if `ipPools` is empty will just return.